packages/contracts-bedrock/contracts/L1/SystemConfig.sol

```diff
@@ -29,6 +29,36 @@ contract SystemConfig is OwnableUpgradeable, Semver {
         UNSAFE_BLOCK_SIGNER
     }
 
+    /**
+     * @notice Represents the configuration for the EIP-1559 based curve for
+     *         the deposit gas market. These values should be set with care
+     *         as it is possible to set them in a way that breaks the deposit
+     *         gas market. The target resource limit is defined as
+     *         maxResourceLimit / elasticityMultiplier.
+     *         This struct was designed to fit within a single word. There is
+     *         additional space for additions in the future.
+     *
+     * @custom:field maxResourceLimit             Represents the maximum amount of deposit
+     *                                            gas that can be purchased per block.
+     * @custom:field elasticityMultiplier         Determines the target resource limit
+     *                                            along with the resource limit.
+     * @custom:field baseFeeMaxChangeDenominator  Determines max change on fee per block.
+     * @custom:field minimumBaseFee               The min deposit base fee, it is clamped to this
+     *                                            value.
+     * @custom:field systemTxMaxGas               The amount of gas supplied to the system
+     *                                            transaction.
+     * @custom:field maximumBaseFee               The max deposit base fee, it is clamped to this
+     *                                            value.
+     */
+    struct ResourceConfig {
+        uint32 maxResourceLimit;
+        uint8 elasticityMultiplier;
+        uint8 baseFeeMaxChangeDenominator;
+        uint32 minimumBaseFee;
+        uint32 systemTxMaxGas;
```

Comment:
We need to document that this parameter needs to be increased if you are adding new system txns in the stack or if the gas given to a system txn changes.

Reply:
ideally in the long term, changes to this can be used to pick how much gas is supplied to the system tx

Reply:
added docs here about it